### PR TITLE
KSE-1795 | Add support for AnyOf Json type while encoding Ksql schemas.

### PR DESCRIPTION
**Pull request by pbadani**
### Description 

Union/AnyOf types in Schema Registry are encoded in Connect as a Struct with a special schema name: **io.confluent.connect.json.OneOf** and one field for each of the alternative types.

{"anyOf":[{"type":"string"},{"type":"integer"},{"type":"boolean"}]

The fields for each alternative types are named as following for the given example:
**io.confluent.connect.json.OneOf.field.0** - String type
**io.confluent.connect.json.OneOf.field.1** - Integer type
**io.confluent.connect.json.OneOf.field.2** - boolean type

During Ser/De in the JsonSchemaConverter the first field with a non-null value is picked as the value of the current Union/AnyOf type. (see Class JsonSchemaData in Schema Registry).

KsqlDB schema in the Command object does not maintain this special information for the Structs (nor does it store the names associated with the field schemas) and hence it ends up treating it as a regular struct thereby changing its interpretation during serialization. This results in Schema mismatch errors.

To handle this, If all the fields in this struct follow the above naming convention, we infer this Struct to be of a Union type.

While preparing the Connect record for Serialization, we explicitly set the schema name as io.confluent.connect.json.OneOf if the current Struct is of Union type.
see io.confluent.ksql.schema.ksql.SchemaConverters

### Testing done 
Tested these changes via docker image built locally
1. Register a schema with anyOf type in SR.
`{"schemaType":"JSON", "schema": "{"additionalProperties":false,"properties":{"name":{"oneOf":[{"type":"null"},{"type":"string"}]},"value":{"anyOf":[{"type":"null"},{"type":"object"},{"type":"string"},{"type":"integer"},{"type":"boolean"}]},"valueType":{"oneOf":[{"type":"null"},{"type":"string"}]}},"type":"object"}"}`

2. Produce records to a topic with this schema
`kafka-json-schema-console-producer --bootstrap-server ksql_kafka_1:9092 --topic test-value --property value.schema.id=1`

3. Create a stream with the schema id.
`CREATE STREAM test1 WITH (kafka_topic='test-value',value_format='json_sr',value_schema_id=1);`

4. Create another stream from the previous stream
`CREATE STREAM another as SELECT * FROM test1;`

This registers a new schema (oneOf with multiple alternatives)
`{"type":"object","properties":{"valueType":{"connect.index":0,"oneOf":[{"type":"null"},{"type":"string"}]},"name":{"connect.index":1,"oneOf":[{"type":"null"},{"type":"string"}]},"value":{"connect.index":2,"oneOf":[{"type":"null"},{"type":"object","connect.index":0},{"type":"string","connect.index":1},{"type":"integer","connect.index":2,"connect.type":"int64"},{"connect.index":3,"type":"boolean"}]}}}`

5. Create another stream with explicit schema id (doesn't register a new schema)
`CREATE STREAM AnotherWithSchema WITH (value_schema_id=1) as SELECT * FROM test1;`

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")
- [ ] Do these changes have compatibility implications for rollback? If so, ensure that the ksql [command version](https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java#L41) is bumped.

### CODE REVIEW

File: ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SchemaConverters.java [hunk 1]
```diff
@@ -344,8 +346,21 @@ private static SchemaBuilder fromSqlStruct(final SqlStruct struct) {
...
-          .optional();
+      Optional<SqlStruct.UnionType> unionType = struct.unionType();
+
+      if (unionType.isPresent()) {
+        switch (unionType.get()) {
+          case ONE_OF_TYPE:
+            builder.name(JsonSchemaData.JSON_TYPE_ONE_OF);
+            break;
+          case GENERALIZED_TYPE:
+            builder.parameter(JsonSchemaData.GENERALIZED_TYPE_UNION, "0");
```
Comment (pbadani) [inline ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SchemaConverters.java:357]: https://github.com/confluentinc/schema-registry/blob/99fc46b6a39b77c39ab72c44f405f7a856a1d28c/json-schema-converter/src/main/java/io/confluent/connect/json/JsonSchemaData.java#L1292-L1294

File: ksqldb-common/src/test/java/io/confluent/ksql/schema/ksql/SchemaConvertersTest.java [hunk 2]
```diff
@@ -309,6 +317,84 @@ public void shouldConvertJavaStringToSqlTimestamp() {
...
+        .field("io.confluent.connect.json.OneOf.field.1", SqlPrimitiveType.of(SqlBaseType.BOOLEAN))
+        .field("io.confluent.connect.json.OneOf.field.2", SqlPrimitiveType.of(SqlBaseType.INTEGER))
+        .build();
+
+    // When:
+    Schema connectSchema = sqlToConnectConverter().toConnectSchema(sqlStruct);
+
+    // Then:
+    assertThat(connectSchema.type(), is(Schema.Type.STRUCT));
+    assertEquals(JSON_TYPE_ONE_OF, connectSchema.schema().name());
```
Comment (pbadani) [inline ksqldb-common/src/test/java/io/confluent/ksql/schema/ksql/SchemaConvertersTest.java:334]: Assert for the name of the schema.

File: ksqldb-common/src/test/java/io/confluent/ksql/schema/ksql/SchemaConvertersTest.java [hunk 3]
```diff
@@ -309,6 +317,84 @@ public void shouldConvertJavaStringToSqlTimestamp() {
...
+        .field("connect_union_field_2", SqlPrimitiveType.of(SqlBaseType.INTEGER))
+        .build();
+
+    // When:
+    Schema connectSchema = sqlToConnectConverter().toConnectSchema(sqlStruct);
+
+    // Then:
+    assertThat(connectSchema.type(), is(Schema.Type.STRUCT));
+    assertFalse(connectSchema.schema().parameters().isEmpty());
+    assertTrue(connectSchema.schema().parameters().containsKey(JsonSchemaData.GENERALIZED_TYPE_UNION));
```
Comment (pbadani) [inline ksqldb-common/src/test/java/io/confluent/ksql/schema/ksql/SchemaConvertersTest.java:352]: Assert that the schema parameters contains an entry for the generalized type.

**Review by rayokota — APPROVED** [review]
Thanks @pbadani , LGTM

**Pull request merged**